x/interchainqueries/keeper/keeper.go

```diff
@@ -260,3 +263,26 @@ func (k Keeper) checkRegisteredQueryExists(ctx sdk.Context, id uint64) bool {
 
 	return store.Has(types.GetRegisteredQueryByIDKey(id))
 }
+
+func (k Keeper) CollectDeposit(ctx sdk.Context, queryInfo types.RegisteredQuery) error {
+	owner, err := queryInfo.GetOwnerAddress()
+	if err != nil {
+		panic(err.Error())
```

Comment:
I don't think `panic` is good here. Why not just return an error like everywhere else?